From: Saeed Bishara <saeed@marvell.com>
Date: Tue, 27 Jan 2009 18:15:59 +0200
Subject: [PATCH] sata: increate the IORDY timeout for the soc controllers

the old value (in clock cycles) is not suitable for soc devices that has
internal clock more that 150MHz.

Signed-off-by: Saeed Bishara <saeed@marvell.com>


--- a/drivers/ata/sata_mv.c	2009-04-11 15:40:49.000000000 +0000
+++ b/drivers/ata/sata_mv.c	2009-04-11 15:40:54.000000000 +0000
@@ -2662,7 +2662,7 @@
 	ZERO(0x024);		/* respq outp */
 	ZERO(0x020);		/* respq inp */
 	ZERO(0x02c);		/* test control */
-	writel(0xbc, port_mmio + EDMA_IORDY_TMOUT);
+	writel(0x800, port_mmio + EDMA_IORDY_TMOUT);
 }
 
 #undef ZERO
